R/ExtractData_Soils.R

```diff
@@ -263,6 +263,212 @@ extract_soil_CONUSSOIL <- function(MMC, sim_size, sim_space, dir_ex_soil,
+          file_type <- "matricd";
+        }
+        else if (grepl("^gravel", tif_file)){
+          # switch file_type to match with MMC for insertion
+          file_type <- "GravelContent"
+        }
+        else{
+          file_type <- substr(tif_file, 1, regexpr("_", tif_file) - 1);
+        }
+        # find the layer number of the current file ===================
+        ltemp <- strsplit(tif_file, "");
+        found_layer <- FALSE;
+        i <- 1;
+        while (!found_layer){
+          if (grepl("[[:digit:]]", ltemp[[1]][i])){
+            soil_layer <- type.convert(ltemp[[1]][i]) - 1;
+            found_layer <- TRUE;
+          }
+          else{
+            i <- i + 1;
+          }
+        }
+        # start extracton process =============================
+        MMC[["idone"]]["GriddedFROM250m"] <- FALSE
+        n_extract <- sum(todos)
+        if (n_extract > 0) {
+          if (show_site_info){
+            print(paste("Soils data from 'Gridded250m' will be extracted for",
+                        "n =", n_extract, "sites"))
+            show_site_info <- FALSE
+          }
+          tif_file <- paste0(paste0(dir.ex.gridded, "/"), tif_file);
+          if (!file.exists(tif_file)) stop(paste0("File '", tif_file,
+                                                 "' does not exist"))
+          layer_N <- length(ldepth_gridded) - 1
+          ils <- seq_len(layer_N)
+          g <- raster::brick(tif_file)
+          soil_data <- raster::crs(g)
+
+          if (!raster::compareCRS(sim_space[["crs_sites"]], soil_data)) {
+            #transform graphics::points to grid-coords
+            sites_conus <- sp::spTransform(sites_conus, CRS = soil_data)
+          }
+          if (sim_space[["scorp"]] == "point") {
+            cell_res_conus <- NULL
+            args_extract <- list(y = sites_conus, type = sim_space[["scorp"]])
+          } else if (sim_space[["scorp"]] == "cell") {
+            cell_res_conus <- align_with_target_res(
+                                        res_from = sim_space[["sim_res"]],
+                                        crs_from = sim_space[["sim_crs"]],
+                                        sp = sim_space[["run_sites"]][todos, ],
+                                        crs_sp = sim_space[["crs_sites"]],
+                                        crs_to = soil_data)
+            args_extract <- list(y = cell_res_conus, coords = sites_conus,
+                                 method = "block", type = sim_space[["scorp"]])
+          }
+          # actually extract the sand values from the sites
+          cond30 <- compiler::cmpfun(function(v) ifelse(is.na(v) | v < 30, NA,
+                                                        v))
+          ftemp <- tif_file
+          g <- if (file.exists(ftemp)) {
+            raster::brick(ftemp)
+          } else {
+            raster::calc(g, fun = cond30, filename = ftemp)
+          }
+          # get max depths for site soil profiles
```

Comment:
We are here inside the `for` loop, i.e., you extract soil depth values for each of the other files -- again and again??